grails-app/services/org/pih/warehouse/core/IdentifierService.groovy

```diff
@@ -261,9 +271,61 @@ class IdentifierService {
         return renderTemplate(template, model)
     }
 
+    String generateSequentialProductIdentifierForDefaultProductType() {
+        Integer sequenceNumber = getNextSequentialProductCodeForDefaultType()
+        String sequenceNumberStr = generateSequenceNumber(sequenceNumber.toString())
+        Map model = ["sequenceNumber": sequenceNumberStr, "productTypeCode": '', "delimiter": '']
+
+        String template = ConfigurationHolder.config.openboxes.identifier.productCode.format
+        return renderTemplate(template, model)
+    }
+
+    def getCharacterOptions(String character) {
+        def characterOptions
+        switch (character) {
+            case '0':
+                characterOptions = ConfigurationHolder.config.openboxes.identifier.numeric
+                break
+            case 'N':
+                characterOptions = ConfigurationHolder.config.openboxes.identifier.numeric
+                break
+            case 'D':
+                characterOptions = ConfigurationHolder.config.openboxes.identifier.numeric
+                break
+            case 'L':
+                characterOptions = ConfigurationHolder.config.openboxes.identifier.alphabetic
+                break
+            case 'A':
+                characterOptions = ConfigurationHolder.config.openboxes.identifier.alphanumeric
+                break
+            default:
+                characterOptions = ''
```

Comment:
we should for now only allow `0, N, and D`, because we do not support alphabetic sequence increments
if we get anything else, then we probably (?) should throw an exception here

Reply:
L and A might be used when we get the "prefix" and delimeter into account